### PR TITLE
Add bgzip input VCF support

### PR DESCRIPTION
**Pull request by jasonwalker80**
Add bgzip support for VCF files. Mostly pulled from Genome::Sys in genome/genome

https://github.com/genome/genome/blob/63da350a240bede3a5415f515b72b0fb6cdb8d01/lib/perl/Genome/Utility/Vcf.pm#L80

### CODE REVIEW

File: fpfilter.pl [hunk 1]
```diff
@@ -603,3 +604,128 @@ sub index_bam {
...
+}
+
+sub open_file_for_reading {
+    my $file = shift;
+
+    validate_file_for_reading($file)
+        or return;
+
+    # _open_file throws its own exception if it doesn't work
+    return _open_file($file, 'r');
```
Comment (tmooney) [inline fpfilter.pl]: Maybe each use of this could be replaced by `IO::File->new($file)` .  `_open_file` seems like overkill for what we're doing here.

Comment (jasonwalker80) [reply fpfilter.pl]: Will `_open_file($pipe)` work the same as `IO::File->new($pipe)`?

Comment (tmooney) [reply fpfilter.pl]: It ought to. `_open_file` ends up on [this line](https://github.com/jasonwalker80/fpfilter-tool/blob/c901a8a168167ae4ad7db56229c3e3ad4de5e902/fpfilter.pl#L690):
```perl
my $fh = (defined $rw) ? IO::File->new($file, $rw) : IO::File->new($file);
```
so we could do `IO::File->new($file, 'r')` to match.

**Review by tmooney — APPROVED** [review]
+1 :film_projector:

**Pull request merged**